### ISSUE TITLE
Golem RSS feed Utf8Error

### ISSUE THREAD

**Issue by jangernert** [feed-rs/feed-rs#10]
When sending a get request with `reqwest` and getting the raw bytes from `https://rss.golem.de/rss.php?feed=RSS1.0` and then feeding them to `feed-rs` yields the following error when parsing the description of the first entry 
```
Err value: Error { pos: 1:1, kind: Utf8(Utf8Error { valid_up_to: 0, error_len: Some(1) }) }
```

Getting the response body with `.text()` works fine.

I prepared a small code sample

```
use std::io::BufReader;

#[tokio::main]
async fn main() {
    let rss_response = reqwest::get("https://rss.golem.de/rss.php?feed=RSS1.0").await.unwrap();
    //let rss_response = rss_response.text().await.unwrap();
    //let feed = feed_rs::parser::parse(rss_response.as_bytes()).unwrap();
    let rss_response = rss_response.bytes().await.unwrap().to_vec();
    let feed = feed_rs::parser::parse(BufReader::new(rss_response.as_slice())).unwrap();
    println!("{:?}", feed.title);
}
```

The description it chokes on is 
```
<description>Envelope, ein Umschlag aus Papier für das eigene Telefon. Allerdings können Nutzer damit den Bildschirm nicht mehr erkennen und nur noch ein Nummernpad und wenige Tasten verwenden. Einen Bastelbogen und die App stellt das Team kostenlos zur Verfügung. (&lt;a href=&quot;https://www.golem.de/specials/smartphone/&quot;&gt;Smartphone&lt;/a&gt;, &lt;a href=&quot;https://www.golem.de/specials/google/&quot;&gt;Google&lt;/a&gt;) &lt;img src=&quot;https://cpx.golem.de/cpx.php?class=17&amp;amp;aid=146212&amp;amp;page=1&amp;amp;ts=1579701600&quot; alt=&quot;&quot; width=&quot;1&quot; height=&quot;1&quot; /&gt;</description>
```

The only difference I can think of is that `reqwest` somehow fixes a malformed sequence in the feed when using [`.text()`](https://docs.rs/reqwest/0.10.1/reqwest/struct.Response.html#method.text).

```
This method decodes the response body with BOM sniffing and with malformed sequences replaced with the REPLACEMENT CHARACTER. Encoding is determinated from the charset parameter of Content-Type header, and defaults to utf-8 if not presented.
```

**Comment by markpritchard** [comment]
Curious, the changes I made in the fix for #7 introduced a little command line tool, which I ran as follows (putting `https://rss.golem.de/rss.php?feed=RSS1.0` in `urls.txt`)

```
cat urls.txt | cargo run --bin testurls | tee ~/tmp/feed-test.log
```

works fine. I wonder if there is something strange with your request pipeline / decoding?

**Comment by markpritchard** [comment]
I'll close this issue for now, but feel free to re-open if you still have problems.
